### PR TITLE
Makefile: misc cleanup

### PR DESCRIPTION
**Pull request by tamird**
Extracted from #13102.

---
This change is [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/13108)

### CODE REVIEW

**Comment by petermattis** [comment]
:lgtm:

---

Review status: 0 of 6 files reviewed at latest revision, 2 unresolved discussions, some commit checks pending.

---

*[Makefile, line 241 at r6](https://reviewable.io:443/reviews/cockroachdb/cockroach/13108#-KbD4CeVISzg4H0fRKl_:-KbD4CeW-N2yjLDqUe41:b-3stkji) ([raw file](https://github.com/cockroachdb/cockroach/blob/68f7b6c49f9df6849c21eb389ef4b2308b44fc11/Makefile#L241)):*
> ```Makefile
> .bootstrap: $(GITHOOKS) GLOCKFILE glide.lock
> 	git submodule update --init
> 	$(GO) install ./vendor/github.com/robfig/glock
> ```

I noticed this earlier today. Do we still need `glock` for something?

---

*[build/builder.sh, line 110 at r6](https://reviewable.io:443/reviews/cockroachdb/cockroach/13108#-KbD4hH6HT4PiUgdl5wt:-KbD4hH6HT4PiUgdl5wu:b5qq84b) ([raw file](https://github.com/cockroachdb/cockroach/blob/68f7b6c49f9df6849c21eb389ef4b2308b44fc11/build/builder.sh#L110)):*
> ```Shell
> # a pager, so we override $PAGER to disable.
> vols=""
> vols="${vols} --volume=${passwd_file}:/etc/passwd"
> ```

How does this make it easier to run as root?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/13108#-:-KbD4Sqa0hrdsfBc5gCo:bnfp4nl)*

**Comment by dt** [comment]
:lgtm:

---

Review status: 0 of 6 files reviewed at latest revision, 2 unresolved discussions, some commit checks pending.

---

*[Makefile, line 241 at r6](https://reviewable.io:443/reviews/cockroachdb/cockroach/13108#-KbD4CeVISzg4H0fRKl_:-KbD4w_oQVlcx1eRVzOm:b-vctex1) ([raw file](https://github.com/cockroachdb/cockroach/blob/68f7b6c49f9df6849c21eb389ef4b2308b44fc11/Makefile#L241)):*
<details><summary><i>Previously, petermattis (Peter Mattis) wrote…</i></summary><blockquote>

I noticed this earlier today. Do we still need `glock` for something?
</blockquote></details>

We use it as a fancy way to run `go instal ./vendor/blah`, though we could easily switch to direct `go install` calls

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/13108#-:-KbD5HsiHPrPrXUwE9fj:bnfp4nl)*

**Comment by tamird** [comment]
Review status: 0 of 6 files reviewed at latest revision, 2 unresolved discussions, some commit checks pending.

---

*[Makefile, line 241 at r6](https://reviewable.io:443/reviews/cockroachdb/cockroach/13108#-KbD4CeVISzg4H0fRKl_:-KbD53T_woC_mqxMuyDm:b-vty5x8) ([raw file](https://github.com/cockroachdb/cockroach/blob/68f7b6c49f9df6849c21eb389ef4b2308b44fc11/Makefile#L241)):*
<details><summary><i>Previously, petermattis (Peter Mattis) wrote…</i></summary><blockquote>

I noticed this earlier today. Do we still need `glock` for something?
</blockquote></details>

Yeah, this target here; we use glock to invoke `go install` on a bunch of stuff. See the GLOCKFILE.

---

*[build/builder.sh, line 110 at r6](https://reviewable.io:443/reviews/cockroachdb/cockroach/13108#-KbD4hH6HT4PiUgdl5wt:-KbD5FJe3mKf98N5RCzc:b-jbcc4b) ([raw file](https://github.com/cockroachdb/cockroach/blob/68f7b6c49f9df6849c21eb389ef4b2308b44fc11/build/builder.sh#L110)):*
<details><summary><i>Previously, petermattis (Peter Mattis) wrote…</i></summary><blockquote>

How does this make it easier to run as root?
</blockquote></details>

Because the necessary diff becomes
```diff
diff --git a/build/builder.sh b/build/builder.sh
index b5f402189..e5dbfcebb 100755
--- a/build/builder.sh
+++ b/build/builder.sh
@@ -107,7 +107,6 @@ mkdir -p "${host_home}"/.{jspm,yarn-cache}
 # -i causes some commands (including `git diff`) to attempt to use
 # a pager, so we override $PAGER to disable.
 vols=""
-vols="${vols} --volume=${passwd_file}:/etc/passwd"
 vols="${vols} --volume=${host_home}:${container_home}"
 vols="${vols} --volume=${gopath0}/src:/go/src"
 vols="${vols} --volume=${gopath0}/pkg/docker_amd64:/go/pkg/linux_amd64"
@@ -134,7 +133,6 @@ if test -e "${alternates_file}"; then
 fi
 
 docker run -i ${tty-} --rm \
-  -u "${uid_gid}" \
   ${vols} \
   --workdir="/go/src/github.com/cockroachdb/cockroach" \
   --env="TMPDIR=/go/src/github.com/cockroachdb/cockroach/artifacts" \
diff --git a/pkg/acceptance/cluster/localcluster.go b/pkg/acceptance/cluster/localcluster.go
index a2b59098a..8e9582990 100644
--- a/pkg/acceptance/cluster/localcluster.go
+++ b/pkg/acceptance/cluster/localcluster.go
@@ -58,7 +58,7 @@ import (
 
 const (
 	builderImage     = "cockroachdb/builder"
-	builderTag       = "20170123-165351"
+	builderTag       = "latest"
 	builderImageFull = builderImage + ":" + builderTag
 	networkPrefix    = "cockroachdb_acceptance"
 )
```

in other words, you just delete this line, instead of having to replace it with `vols=""` or having to edit the next line.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/13108)*

**Comment by bdarnell** [comment]
:lgtm:

---

Reviewed 2 of 7 files at r7, 3 of 3 files at r8, 1 of 1 files at r12, 1 of 1 files at r13, 1 of 1 files at r14.
Review status: all files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[Makefile, line 95 at r14](https://reviewable.io:443/reviews/cockroachdb/cockroach/13108#-KbDaANR3TcyiosDWb21:-KbDaANSRTKD8lkJuX40:b-xaz55n) ([raw file](https://github.com/cockroachdb/cockroach/blob/89d4d5c9d9aa1840a2437af9befdee9965c6bd64/Makefile#L95)):*
> ```Makefile
> 	@echo "$$GOPATH/bin added to PATH"
> 	@echo $(GO) $(BUILDMODE) -v $(GOFLAGS) -tags '$(TAGS)' $(BUILDTARGET)
> 	@$(GO) $(BUILDMODE) -v $(GOFLAGS) -tags '$(TAGS)' -ldflags '$(LDFLAGS)' $(BUILDTARGET)
> ```

Now that `$(LDFLAGS)` is smaller, maybe we should just remove the leading `@` from this line and the `echo` statement that prints a subset of the command.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/13108#-:-KbDaf6hblJrLwP8uSq2:bnfp4nl)*

**Comment by tamird** [comment]
Review status: 2 of 8 files reviewed at latest revision, 2 unresolved discussions.

---

*[Makefile, line 95 at r14](https://reviewable.io:443/reviews/cockroachdb/cockroach/13108#-KbDaANR3TcyiosDWb21:-KbEVzf8qu8qG8pA36za:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/89d4d5c9d9aa1840a2437af9befdee9965c6bd64/Makefile#L95)):*
<details><summary><i>Previously, bdarnell (Ben Darnell) wrote…</i></summary><blockquote>

Now that `$(LDFLAGS)` is smaller, maybe we should just remove the leading `@` from this line and the `echo` statement that prints a subset of the command.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/13108)*
